examples/03-http-any.php

```diff
@@ -22,6 +22,8 @@
         function (Psr\Http\Message\ResponseInterface $response) use ($url) {
             // return only the URL for the first successful response
             return $url;
+        }, function (Exception $e) {
+            echo 'Error: ' . $e->getMessage() . PHP_EOL;
```

Comment:
Looks reasonable, but unfortunately changes the outcome of `$promise` as it may no longer reject, but would instead resolve with a `string` from the fulfillment handler or an implicit `null` from the rejection handler.

I would rather remove this here and update the finial rejection handler near the bottom instead. Similarly, we may want to update the final rejection handlers in the other examples. WDYT?